source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -91,34 +104,20 @@ void SingleServerRequest::onFailure() {
   callbacks_.onResponse(Utility::makeError(Response::get().UpstreamFailure));
 }
 
-void SingleServerRequest::recreate(Common::Redis::RespValue& request, bool prepend_asking) {
-  if (!prepend_asking) {
-    request = *incoming_request_;
-    return;
-  }
-
-  Common::Redis::RespValue asking_cmd;
-  asking_cmd.type(Common::Redis::RespType::BulkString);
-  asking_cmd.asString() = "asking";
-
-  request.type(Common::Redis::RespType::Array);
-  request.asArray().push_back(asking_cmd);
-  request.asArray().insert(request.asArray().end(), incoming_request_->asArray().begin(),
-                           incoming_request_->asArray().end());
-}
-
 bool SingleServerRequest::onRedirection(const Common::Redis::RespValue& value) {
   std::vector<absl::string_view> err;
   bool ask_redirection = false;
   if (redirectionArgsInvalid(incoming_request_.get(), value, err, ask_redirection) || !conn_pool_) {
     return false;
   }
 
-  Common::Redis::RespValue request;
-  recreate(request, ask_redirection);
-
   const std::string host_address = std::string(err[2]); // ip:port
-  handle_ = conn_pool_->makeRequestToHost(host_address, request, *this);
+  // Prepend request with asking command if redirected via ASK error.
+  if (ask_redirection &&
```

Comment:
You could turn this into a function with the below code in `onChildRedirection`, but that's totally optional. It's more readable as is.

Reply:
I agree that it is more readable as it is, but the big explanatory comment in onChildRedirection() should probably also be here. I'll push a change that adds the big explanation comment here as well.